x-pack/plugins/security_solution/server/endpoint/migrations/turn_off_policy_protections.test.ts

```diff
@@ -29,306 +32,431 @@ describe('Turn Off Policy Protections Migration', () => {
   const callTurnOffPolicyProtections = () =>
     turnOffPolicyProtectionsIfNotSupported(esClient, fleetServices, productFeatureService, logger);
 
+  const mockPolicyListResponse = (
+    { total, items, page }: { total?: number; items?: PolicyData[]; page?: number } = {
+      total: 1,
+      page: 2,
+      items: [],
+    }
+  ) => {
+    const packagePolicyListSrv = fleetServices.packagePolicy.list as jest.Mock;
+    return packagePolicyListSrv.mockResolvedValueOnce({
+      total,
+      page,
+      perPage: 1500,
+      items,
+    });
+  };
+
   const generatePolicyMock = (
-    policyGenerator: FleetPackagePolicyGenerator,
     withDisabledProtections = false,
-    withDisabledProtectionUpdates = true
+    withCustomProtectionUpdates = true,
+    withCustomNotifications = true
   ): PolicyData => {
-    const policy = policyGenerator.generateEndpointPackagePolicy();
-
-    if (!withDisabledProtections && withDisabledProtectionUpdates) {
-      return policy;
-    } else if (!withDisabledProtections && !withDisabledProtectionUpdates) {
-      policy.inputs[0].config.policy.value.global_manifest_version = '2023-01-01';
-      return policy;
-    } else if (withDisabledProtections && !withDisabledProtectionUpdates) {
+    const policy = new FleetPackagePolicyGenerator('seed').generateEndpointPackagePolicy();
+    if (withDisabledProtections) {
       policy.inputs[0].config.policy.value = ensureOnlyEventCollectionIsAllowed(
         policy.inputs[0].config.policy.value
       );
+    }
+    if (withCustomProtectionUpdates) {
       policy.inputs[0].config.policy.value.global_manifest_version = '2023-01-01';
-      return policy;
-    } else {
-      policy.inputs[0].config.policy.value = ensureOnlyEventCollectionIsAllowed(
-        policy.inputs[0].config.policy.value
+    }
+    if (!withCustomNotifications) {
+      policy.inputs[0].config.policy.value = merge(
+        {},
+        policy.inputs[0].config.policy.value,
+        resetCustomNotifications()
+      );
+    } else if (withCustomNotifications) {
+      policy.inputs[0].config.policy.value = merge(
+        {},
+        policy.inputs[0].config.policy.value,
+        resetCustomNotifications('custom test')
       );
-      return policy; // This is the only one that shouldn't be updated since it has default values for disabled features
     }
+    return policy;
   };
 
+  const generateExpectedPolicyMock = ({
```

Comment:
I decided to keep this fragment hardcoded, as any attempts to generate it with iterations resulted in code that, in my opinion, would be difficult to read and hard for anyone to understand.